src/Exceptions/InvalidFilterProperty.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Spatie\QueryBuilder\Exceptions;
+
+use Exception;
+
+class InvalidFilterProperty extends Exception
+{
+    public static function make($property): static
+    {
+        return new static("Filter property `{$property}` is invalid.");
```

Comment:
Can we add a `$reason` to this exception? From the implementation I see there's only one reason this happens: `belongs to relationship is not an Eloquent relationship`.

Reply:
Exactly.
There was another exception that I wasn't handling (on purpose), when the relation (or the last relation in a nested dotted string) is not a BelongTo relation, in that case an unhandled exception was thrown but Laravel gave a correct message and context to find this error. This case was not covered in the tests.
Based on that exception, now remove `InvalidFilterProperty` and throw that same exception `RelationNotFoundException`. I also added more test cases that cover all the options (this case and the previous one).
This native Laravel exception gives a more specific message to fix the error.

Reply:
@AlexVanderbist Have you been able to review this?

Reply:
@freekmurze Have you been able to review this?

Reply:
Perfect, thank you!